main/src/library/MutHashSet.flix

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright 2025 Magnus Madsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// Represents a mutable hash set that preserves insertion order.
+///
+/// Uses a MutHashMap internally, storing elements as keys with Unit values.
+///
+pub struct MutHashSet[t: Type, r: Region] {
+    inner: MutHashMap[t, Unit, r]
+}
+
+instance ForEach[MutHashSet[a, r]] {
+    type Elm = a
+    type Aef = r
+    pub def forEach(f: a -> Unit \ ef, s: MutHashSet[a, r]): Unit \ ef + r =
+        MutHashSet.forEach(f, s)
+}
+
+instance Iterable[MutHashSet[a, r]] {
+    type Elm = a
+    type Aef = r
+    pub def iterator(rc: Region[r1], s: MutHashSet[a, r]): Iterator[a, r + r1, r1] \ (r + r1) =
+        MutHashSet.iterator(rc, s)
+}
+
+mod MutHashSet {
+
+    ///
+    /// Returns a fresh empty set.
```

Comment:
It's consistent with the other collections but its funny that we call it fresh here. And `emptyWithCapacity` is not fresh i guess